packages/iov-keycontrol/src/wallets/slip10wallet.ts

```diff
@@ -376,16 +383,27 @@ export class Slip10Wallet implements Wallet {
     return derivationResult.privkey;
   }
 
-  private buildLocalIdentity(bytes: PublicKeyBytes, label: string | undefined): LocalIdentity {
+  private buildLocalIdentity(
+    chainId: ChainId,
+    bytes: PublicKeyBytes,
+    label: string | undefined,
+  ): LocalIdentity {
+    if (!chainId) {
+      throw new Error("Got empty chain ID when tying to build a local identity.");
+    }
+
     const algorithm = Slip10Wallet.algorithmFromCurve(this.curve);
-    const pubkey: PublicKeyBundle = {
-      algo: algorithm,
-      data: bytes,
+    const publicIdentity: PublicIdentity = {
+      chainId: chainId,
+      pubkey: {
+        algo: algorithm,
+        data: bytes,
+      },
     };
     return {
-      pubkey,
-      label,
-      id: Slip10Wallet.identityId({ pubkey }),
+      ...publicIdentity,
+      label: label,
+      id: Slip10Wallet.identityId(publicIdentity),
```

Comment:
I don't see changes to the `identityId` function, meaning it is the same regardless of chainId?

If this becomes an intrinsic part, we should add that to the `id` as well, right? (Especially since we are breaking any db migrations this time, let's break it even more)

Reply:
Great catch! This is a horrible bug